packages/components/src/inputs/src/InputGroup.module.css

```diff
@@ -0,0 +1,124 @@
+.hop-InputGroup  {
+    --hop-InputGroup-background: var(--hop-neutral-surface);
+    --hop-InputGroup-border-color: var(--hop-neutral-border);
+    --hop-InputGroup-border-width: 0.0625rem;
+    --hop-InputGroup-gap: var(--hop-space-inline-sm);
+    --hop-InputGroup-sm-block-size: 2rem;
+    --hop-InputGroup-md-block-size: 2.5rem;
+    --hop-InputGroup-border-radius: var(--hop-shape-rounded-md);
+    --hop-InputGroup-padding-inline: var(--hop-space-inset-md) var(--hop-space-inset-md);
+    --hop-InputGroup-border-color-hover: var(--hop-neutral-border-hover);
+    --hop-InputGroup-border-color-focus: var(--hop-primary-border-focus);
+    --hop-InputGroup-background-hover: var(--hop-neutral-surface-hover);
+    --hop-InputGroup-border-color-invalid: var(--hop-danger-border-strong);
+    --hop-InputGroup-input-color: var(--hop-neutral-text);
+    --hop-InputGroup-input-sm-font-family: var(--hop-body-sm-font-family);
+    --hop-InputGroup-input-sm-font-size: var(--hop-body-sm-font-size);
+    --hop-InputGroup-input-sm-font-weight: var(--hop-body-sm-font-weight);
+    --hop-InputGroup-input-sm-line-height: var(--hop-body-sm-line-height);
+    --hop-InputGroup-input-sm-block-size: 1.5rem;
+    --hop-InputGroup-input-md-font-family: var(--hop-body-md-font-family);
+    --hop-InputGroup-input-md-font-size: var(--hop-body-md-font-size);
+    --hop-InputGroup-input-md-font-weight: var(--hop-body-md-font-weight);
+    --hop-InputGroup-input-md-line-height: var(--hop-body-md-line-height);
+    --hop-InputGroup-input-md-block-size: 2rem;
+    --hop-InputGroup-background-disabled: var(--hop-neutral-surface-disabled);
+    --hop-InputGroup-border-color-disabled: var(--hop-neutral-border-disabled);
+    --hop-InputGroup-placeholder-font-family: inherit;
+    --hop-InputGroup-placeholder-font-size: inherit;
+    --hop-InputGroup-placeholder-font-weight: inherit;
+    --hop-InputGroup-placeholder-line-height: inherit;
+    --hop-InputGroup-placeholder-color: var(--hop-neutral-text-weakest);
+    --hop-InputGroup-placeholder-color-disabled: var(--hop-neutral-text-disabled);
+
+    /* Internal Variables */
+    --inline-size: fit-content;
+    --border: var(--hop-InputGroup-border-width) solid var(--hop-InputGroup-border-color);
+    --background: var(--hop-InputGroup-background);
+    --block-size: var(--hop-InputGroup-md-block-size);
+
+    position: relative;
+
+    display: inline-flex;
+    gap: var(--hop-InputGroup-gap);
+    align-items: center;
+
+    box-sizing: border-box;
+    inline-size: var(--inline-size);
+    block-size: var(--block-size);
+    padding-inline: var(--hop-InputGroup-padding-inline);
+
+    background: var(--background);
+    border: var(--border);
+    border-radius: var(--hop-InputGroup-border-radius)
+}
+
+.hop-InputGroup--fluid {
+    --inline-size: 100%
+}
+
+.hop-InputGroup--fluid .hop-InputGroup__input {
+    inline-size: 100%;
```

Comment:
I'd say input css should be with the other input css. You can however set an internal variable called --input-inline-size and do like you did with --inline-size. I think it'll clean things up.